### PR TITLE
Spring data jpa

### PR DESCRIPTION
**Pull request by vrudkovsky**

### CODE REVIEW

File: src/main/java/com/vpcomtek/onlinebookstore/model/Book.java [hunk 1]
```diff
@@ -7,23 +7,37 @@
...
 import java.math.BigDecimal;
 import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
 
 @Entity
 @Data
+@SQLDelete(sql = "UPDATE book SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
+//@Table(name = "book")
```
Comment (okuzan) [inline src/main/java/com/vpcomtek/onlinebookstore/model/Book.java]: don't forget to remove comments before committing
```suggestion
@SQLDelete(sql = "UPDATE book SET is_deleted = TRUE WHERE id = ?")
@SQLRestriction(clause = "is_deleted = FALSE")
```

File: src/main/resources/db/changelog/changes/01-create-table-book.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                  name: price
+                  type: decimal
+                  constraints:
+                    nullable: false
+              - column:
+                  name: description
+                  type: varchar(255)
+              - column:
+                  name: cover_image
+                  type: varchar(255)
```
Comment (okuzan) [inline src/main/resources/db/changelog/changes/01-create-table-book.yaml]: don't forget to leave last line of the file empty
image

File: src/main/resources/db/changelog/changes/02-add-column-isDeleted.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+      changes:
+        - addColumn:
+            tableName: book
+            columns:
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (okuzan) [inline src/main/resources/db/changelog/changes/02-add-column-isDeleted.yaml]:
```suggestion
                    nullable: false

```

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+databaseChangeLog:
+  - include:
+      file: db/changelog/changes/01-create-table-book.yaml
+  - include:
+      file: db/changelog/changes/02-add-column-isDeleted.yaml
```
Comment (okuzan) [inline src/main/resources/db/changelog/db.changelog-master.yaml]:
```suggestion
      file: db/changelog/changes/02-add-column-isDeleted.yaml

```

**Review by okuzan — CHANGES_REQUESTED** [review]
extract /api to project-wide prefix using relevent property in application.properties

File: src/main/java/com/vpcomtek/onlinebookstore/model/Book.java (hunk 1 above)
Comment (okuzan) [reply src/main/java/com/vpcomtek/onlinebookstore/model/Book.java]: not fixed

**Review by Vasul-Pankiv — APPROVED** [review]

**Pull request merged**